lib/active_merchant/billing/gateways/global_collect.rb

```diff
@@ -250,20 +255,52 @@ def add_payment(post, payment, options)
         month = format(payment.month, :two_digits)
         expirydate = "#{month}#{year}"
         pre_authorization = options[:pre_authorization] ? 'PRE_AUTHORIZATION' : 'FINAL_AUTHORIZATION'
-        post['cardPaymentMethodSpecificInput'] = {
+        specifics_inputs = {
           'paymentProductId' => BRAND_MAP[payment.brand],
           'skipAuthentication' => 'true', # refers to 3DSecure
           'skipFraudService' => 'true',
           'authorizationMode' => pre_authorization
         }
-        post['cardPaymentMethodSpecificInput']['requiresApproval'] = options[:requires_approval] unless options[:requires_approval].nil?
+        specifics_inputs['requiresApproval'] = options[:requires_approval] unless options[:requires_approval].nil?
+
+        if payment.is_a?(NetworkTokenizationCreditCard) || options[:google_customer_id]
+          source = options[:google_customer_id] ? :google_pay : payment.source
+          specifics_inputs['paymentProductId'] = BRAND_MAP[source]
+          post['mobilePaymentMethodSpecificInput'] = specifics_inputs
+          add_decrypted_payment_data(post, payment, expirydate, source)
+          post['order']['customer']['merchantCustomerId'] = options[:google_customer_id] if options[:google_customer_id]
+        elsif payment.is_a?(CreditCard)
+          post['cardPaymentMethodSpecificInput'] = specifics_inputs
+          post['cardPaymentMethodSpecificInput']['card'] = {
+            'cvv' => payment.verification_value,
+            'cardNumber' => payment.number,
+            'expiryDate' => expirydate,
+            'cardholderName' => payment.name
+          }
+        end
+      end
 
-        post['cardPaymentMethodSpecificInput']['card'] = {
-          'cvv' => payment.verification_value,
-          'cardNumber' => payment.number,
-          'expiryDate' => expirydate,
-          'cardholderName' => payment.name
-        }
+      def add_decrypted_payment_data(post, payment, expirydate, source)
+        data = {}
```

Comment:
❓ Question:

I'm trying to think of a possibility when data is not going to be assigned that justifies this default value, do you know a case?